pandas/tests/extension/test_arrow.py

```diff
@@ -2296,6 +2296,16 @@ def test_str_split_pat_none(method):
     tm.assert_series_equal(result, expected)
 
 
+def test_str_split_regex_none():
```

Comment:
Can you move this test to `pandas/tests/strings/test_split_partition.py`, so we can parametrize this with all the different string dtype implementations, ensuring the different ones all behave the same?

Reply:
Done in the new commit.
But the tests look a bit ugly, because the expected output of `pd.ArrowDtype(pa.string())` has different array dtype from the cases of other string dtypes. Maybe it's better to keep the test separate in `test_arrow.py`?